sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -5854,7 +5854,20 @@ class AstBuilder extends DataTypeAstBuilder
         windowClause = null,
         relation = left,
         isPipeOperatorSelect = true)
-    }.get
+    }.getOrElse(Option(ctx.whereClause).map { c =>
+      // Add a table subquery boundary between the new filter and the input plan if one does not
+      // already exist. This helps the analyzer behave as if we had added the WHERE clause after a
+      // table subquery containing the input plan.
```

Comment:
This is great! This skips the tricky aggregate function pushdown stuff from Filter/Sort which complicates the analyzer quite a bit. We also don't need this with pipe syntax, as it's quite easy for users to filter on the aggregated query.

Reply:
that being said, seems like we don't need to add subquery alias if the child plan is `UnresolvedRelation`. We don't need to isolate the table scan node here.

Reply:
We talked offline and found that updating the `UnresolvedRelation` pattern match to this fixes the problem:

```
        case u: UnresolvedRelation =>
          u
```

In this way we don't add another redundant `SubqueryAlias` when `ResolveRelations` will already add one. Looking at the [commit](https://github.com/apache/spark/pull/48091/commits/0d32c2a23ac7e7f7a5e9f32a4bfb69a802c0bd6c) that performs this update, we see the analyzer plans improve accordingly.

Reply:
ah, it also fixes a regression. We can add a test for `table t |> where spark_catalog.default.t.x = 1`, which didn't work before this fix.

Reply:
Sounds good, this is done.